### PR TITLE
Add option to apply focus styles without :focus.

### LINKED ISSUE #3472: ComboBox: Options need focus styles
### Bug Report

- __Package version(s)__: (fill this out)
- __Browser and OS versions__: (fill this out if relevant)

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? (Yes, No) Maybe

Requested priority: (Blocking, High, Normal, Low) Normal

Products/sites affected: (if applicable)

#### Describe the issue:

In ComboBox.styles.ts optionStyles, `getFocusStyle(theme)` is getting called, but no focus style appears in the browser. IComboBoxOptionStyles extends IButtonStyles so there maybe some conflict there?

#### Actual behavior:

No focus styles / outlines are applied to ComboBox

#### Expected behavior:

Focused options should be outlined 
image

### If applicable, please provide a codepen repro:

https://codepen.io/lynamemi/pen/LOgRRW
Open ComboBox - up and down arrows move focus, that's when the focus outline should appear.

### PR DESCRIPTION
**Pull request by jordandrako**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #3472
- [x] Include a change request file using `$ npm run change`

#### Description of changes

The options in ComboBox don't actually receive keyboard focus, getFocusStyle() doesn't work; it only applies an object of styles to `&:focus:after`. This adds a boolean you can opt into the remove the `:focus` part of the selector, and applies it to ComboBox.

(Apologies for the messy commits and such from #5395)

### CODE REVIEW

File: packages/styling/src/styles/getFocusStyle.ts [hunk 1]
```diff
@@ -22,7 +23,8 @@ export function getFocusStyle(
   position: 'relative' | 'absolute' = 'relative',
   highContrastStyle: IRawStyle | undefined = undefined,
   borderColor: string = theme.palette.white,
-  outlineColor: string = theme.palette.neutralSecondary
+  outlineColor: string = theme.palette.neutralSecondary,
+  isFocusedOnly: boolean = true
```
Comment (JasonGore) [inline packages/styling/src/styles/getFocusStyle.ts]: 👍

**Comment by jordandrako** [comment]
The issue in screener is strange, as I didn't touch the dropdown code, nor is it reproing locally. It's also appearing in #5416.

**Comment by lynamemi** [comment]
You might need to merge master again - I feel like screener sometimes has a timing problem recently merged PRs that had screener changes. I found at least one (semi) recently closed PR relating to Dropdown: https://github.com/OfficeDev/office-ui-fabric-react/pull/5385

**Comment by jordandrako** [comment]
@phkuo Looks like we're waiting on your to review.

**Comment by phkuo** [comment]
Random feedback:
1) We should start moving to semantic slots where possible, especially for common core controls like this one.
2) With all those undefineds... in the future we should consider just taking an 'options' object that has a bunch of optional props. Makes it easy to add more properties later.

**Comment by jordandrako** [comment]
@phkuo I agree. I'll make a feature request to get it on the project tracker.

**Pull request merged**